### PR TITLE
[Fix](planner) Fix wrong planner with count(*) optmizer for cross join optimization

### PR DESCRIPTION
**Pull request by Kikyou1997**
# Proposed changes

Issue Number: noissue

## Problem summary

1. Set source expr for the slotDescrptor of inlineView, so that we could mark the slot in the lower level as materialzied when we do the column prune in the logic of `materializeInlineViewResultExprForCrossJoinOrCountStar`
2. Delay the deletion for not materialized slot in the resolvedTupleExprs in the SortNode, so that we could make sure those slot which would be materialized in the logic of `materializeInlineViewResultExprForCrossJoinOrCountStar` could be add to the resolvedTupleExprs as the output of the SortNode
3. Remove some useless field in the AggregateInfo which always bother and make developer confused when reading code

Besides, this PR will also change the nullability inference of the CreateTableAsSelectStmt, since the nullability of created table is determined by the srcExpr first, if there doesn't exist srcExpr, then will use the nullability of the SlotRef itself, in the past, this srcExpr of tuple created for InlineView is always null, but now we set it.
You can learn these affected logic in the below methods:

* InternalDataSource# createTableAsSelect
* Expr#getSrcSlotRef

## Checklist(Required)

1. Does it affect the original behavior: 
    - [x] Yes
    - [ ] No
    - [ ] I don't know
4. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
5. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
6. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
7. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Kikyou1997** [comment]
@morrySnow @morningman

File: fe/fe-core/src/main/java/org/apache/doris/analysis/SlotDescriptor.java [hunk 1]
```diff
@@ -162,6 +162,21 @@ public void setIsMaterialized(boolean value) {
         isMaterialized = value;
     }
 
+    public void materializeSrcExpr() {
+        if (sourceExprs == null) {
+            return;
+        }
+        for (Expr expr : sourceExprs) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/SlotDescriptor.java:169]: should we need to do chain replace? such as set `IsMaterialized` on the source expr of source expr's slot

File: fe/fe-core/src/main/java/org/apache/doris/planner/SingleNodePlanner.java [hunk 2]
```diff
@@ -1031,9 +1031,33 @@ private PlanNode createSelectPlan(SelectStmt selectStmt, Analyzer analyzer, long
 
             for (int i = 1; i < selectStmt.getTableRefs().size(); ++i) {
                 TableRef innerRef = selectStmt.getTableRefs().get(i);
+                boolean aggOutputNotHaveMaterializedSlot = false;
+                AggregateInfo aggregateInfo = null;
+                TupleDescriptor output = null;
+                if (innerRef instanceof InlineViewRef) {
+                    InlineViewRef inlineViewRef = (InlineViewRef) innerRef;
+                    QueryStmt queryStmt = inlineViewRef.getViewStmt();
+                    if (queryStmt instanceof SelectStmt) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/planner/SingleNodePlanner.java:1040]: set operation do not has this bug?

File: fe/fe-core/src/main/java/org/apache/doris/planner/SortNode.java [hunk 3]
```diff
@@ -211,9 +211,6 @@ public void init(Analyzer analyzer) throws UserException {
         outputSmap = new ExprSubstitutionMap();
 
         for (int i = 0; i < slotExprs.size(); ++i) {
-            if (!sortTupleSlots.get(i).isMaterialized()) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/planner/SortNode.java:214]: add a comment that we rely on projectionPlanner to get a correct plan.

File: fe/fe-core/src/main/java/org/apache/doris/analysis/SlotDescriptor.java (hunk 1 above)
Comment (Kikyou1997) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/SlotDescriptor.java:169]: Yes, we do need

File: fe/fe-core/src/main/java/org/apache/doris/planner/SingleNodePlanner.java (hunk 2 above)
Comment (Kikyou1997) [reply fe/fe-core/src/main/java/org/apache/doris/planner/SingleNodePlanner.java:1040]: I think this bug is just about query block, no matter the stmt of origin SQL is SetOperationStmt or SelectStmt

File: fe/fe-core/src/main/java/org/apache/doris/planner/SortNode.java (hunk 3 above)
Comment (Kikyou1997) [reply fe/fe-core/src/main/java/org/apache/doris/planner/SortNode.java:214]: done

**Review by morrySnow — APPROVED** [review]

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**